app/(saas)/discover/_features/categories/categories.tsx

```
@@ -0,0 +1,48 @@
+import { Section } from "@/app/(saas)/discover/_components/section/section";
+
+import { ProjectCategoryReactQueryAdapter } from "@/core/application/react-query-adapter/project-category";
+
+import { Carousel, CarouselContent, CarouselItem, CarouselNext, CarouselPrevious } from "@/shared/ui/carousel";
+
+export function Categories() {
+  const { data, isLoading, isError } = ProjectCategoryReactQueryAdapter.client.useGetProjectCategories({});
+
+  if (isError) return null;
+
+  const renderCategories = useCallback(() => {
```

ESLint

[error] React Hook "useCallback" is called conditionally. React Hooks must be called in the exact same order in every component render. Did you accidentally call a React Hook after an early return?

(react-hooks/rules-of-hooks)